staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -381,7 +381,7 @@ func (d *DiscoveryClient) ServerVersion() (*version.Info, error) {
 	var info version.Info
 	err = json.Unmarshal(body, &info)
 	if err != nil {
-		return nil, fmt.Errorf("got '%s': %v", string(body), err)
+		return nil, err
```

Comment:
alternative: `error: got '<!DOCTYPE html>...': invalid character '<' looking for beginning of value`

Reply:
i.e. the first line of the body, then ...

Reply:
Ic, you have the error handling outside now

Reply:
I am not a fan of leaking this kind of information (the error type here) outside of a func. Imagine we change the json library or even the protocol. The outer code will break.

Better: handle an arbitrary error in the calling code like:
```
case error:
		klog.V(4).Infof("Server version error: %s", t.Error())
		return fmt.Sprintf("Error retrieving server version: %s", t.Error()), true
```

and then here do: `return nil, fmt.Errorf("cannot parse server response: %s", err)`

Reply:
I see. thanks @sttts

Reply:
@sttts if we returning formatted error in here, it's caused no reason to catching this error to error handling outside. because i'm catching this error by error type (json syntax error) which you think is not good. is this enough to just give an error here?